CMakeLists.txt

```diff
@@ -1,21 +1,124 @@
 cmake_minimum_required(VERSION 3.16.0 FATAL_ERROR) # Configurable policies: <= CMP0097
 
-cmake_policy(SET CMP0091 NEW)
-cmake_policy(SET CMP0092 NEW)
-
-list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules)
-include(LibtorrentMacros)
-
-read_version("${CMAKE_CURRENT_SOURCE_DIR}/include/libtorrent/version.hpp" VER_MAJOR VER_MINOR VER_TINY)
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules")
 
 project(libtorrent
-	DESCRIPTION "Bittorrent library"
-	VERSION ${VER_MAJOR}.${VER_MINOR}.${VER_TINY}
+	VERSION "1.2.14"
+	DESCRIPTION "An efficient feature complete C++ bittorrent implementation"
+	HOMEPAGE_URL "https://libtorrent.org"
+	LANGUAGES C CXX
 )
-set (SOVERSION "10")
 
 include(GNUInstallDirs)
 include(GeneratePkgConfig)
+include(LibtorrentMacros)
+
+# This must be set before creating targets. Ignored on non-MSVC.
+set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>$<$<NOT:$<BOOL:${static_runtime}>>:DLL>")
+
+# -- Features --
+feature_option(NAME build_tests DESCRIPTION "build tests" DEFAULT OFF)
```

Comment:
this is what you mean by "feature summary"?
Does this cause all the options to be printed?
It seems reasonable to me to put them at the top of the file. I take it this isn't declarative, in that lines above the `feature_option` won't actually be able to depend on the option, is that right?

Reply:
@arvidn 

The "feature summary" is this part of the output:


```txt
-- The following features have been enabled:

 * build_examples, build examples (default: OFF)
 * build_tools, build tools (default: OFF)
 * iconv, build libtorrent with iconv support (default: ON)
 * openssl, Build with OpenSSL, to provide HTTPS support to libtorrent (default: ON)
(...)

-- The following features have been disabled:

 * build_tests, build tests (default: OFF)
(...)
```

This is printed by the calls to these functions:

```
feature_summary(DEFAULT_DESCRIPTION WHAT ALL)
feature_summary_multivalue(NAMES MULTIVALUE_FEATURE_NAMES DEFAULTS MULTIVALUE_FEATURE_DEFAULTS)
```

The first one is from the native [FeatureSummary](https://cmake.org/cmake/help/latest/module/FeatureSummary.html) module. The second one is the one I made to print out a summary for multi-valued options as well.

The only restriction is that they must be called **after** all the calls to `feature_option()` and `feature_option_multivalue()`. Because of the way I have structured the build script (so that all feature/option activation is required to be explicit on the part of the user, instead of depending on the environment where the script runs or other features/options), all of said calls are at the very top of the file. As a result, we can put the calls to `feature_summary` pretty much anywhere, not just at the end of the file, which is the usual default.

Reply:
@arvidn 

To be extra clear:

> I take it this isn't declarative, in that lines above the feature_option won't actually be able to depend on the option, is that right?

That's right.